lib/krane/restart_task.rb

```diff
@@ -171,18 +251,38 @@ def fetch_deployments(list)
       end
     end
 
-    def build_patch_payload(deployment)
-      containers = deployment.spec.template.spec.containers
+    def fetch_statefulsets(list)
+      list.map do |name|
+        record = nil
+        begin
+          record = apps_v1_kubeclient.get_stateful_set(name, @namespace)
+        rescue Kubeclient::ResourceNotFoundError
+          raise FatalRestartError, "StatefulSet `#{name}` not found in namespace `#{@namespace}`"
+        end
+        record
+      end
+    end
+
+    def fetch_daemonsets(list)
+      list.map do |name|
+        record = nil
+        begin
+          record = apps_v1_kubeclient.get_daemon_set(name, @namespace)
+        rescue Kubeclient::ResourceNotFoundError
+          raise FatalRestartError, "DaemonSet `#{name}` not found in namespace `#{@namespace}`"
+        end
+        record
+      end
+    end
+
+    def build_patch_payload(_deployment)
       {
         spec: {
           template: {
-            spec: {
-              containers: containers.map do |container|
-                {
-                  name: container.name,
-                  env: [{ name: "RESTARTED_AT", value: Time.now.to_i.to_s }],
```

Comment:
I found an [internal reference to this variable here](https://github.com/Shopify/cloudbuddies/blob/master/buddies/appbuddy/state_config_map.go#L126).

Reply:
This looks like perhaps it was cargo culted from here since they're using it to simply trigger a restart in this case (and they don't particularly care about _reading_ its value). I'm tempted to change the implementation in the buddy to mimic this one (changing annotations on the template metadata). IMO having these processes/controllers mounting variables directly into containers doesn't strike me as the wisest idea.